android-release-build.md

```diff
@@ -0,0 +1,159 @@
+design.
+
+1. In the `<app dir>/android/src/main/res/` directory, place your icon files
+in folders named using [Configuration Qualifiers][configurationqualifiers].
+The default `mipmap-` folders demonstrate the correct naming convention.
+
+1. In `AndroidManifest.xml`, update the [`application`][applicationtag] tag's
+`android:icon` attribute to reference icons from the previous step (e.g.
+`<application android:icon="@mipmap/ic_launcher" ...`).
+
+1. To verify the icon has been replaced, run your app using `flutter run` 
+and inspect the app icon in the Launcher.
+
+## Signing the app
+
+### Create a keystore
+If you have an existing keystore, skip to the next step. If not, create one
+by running the following at the command line:
+`keytool -genkey -v -keystore ~/key.jks -keyalg RSA -keysize 2048 -validity 10000 -alias key`
+
+*Note*: Keep this file private; do not check it into public source control.
+
+### Reference the keystore from the app
+
+Create a file named `<app dir>/android/key.properties` that contains a
+reference to your keystore:
+
+```
+storePassword=<password from previous step>
+keyPassword=<password from previous step>
+keyAlias=key
+storeFile=<location of the key store file, e.g. '/Users/<user name>/key.jks'>
+```
+
+*Note*: Keep this file private; do not check it into public source control.
+
+### Configure signing in gradle
+
+Configure signing for your app by editing the `<app dir>/android/app/build.gradle` 
+file.
+
+1. Replace:
+```
+   android {
+```
+   with the keystore information from your properties file:
+```
+   def keystorePropertiesFile = rootProject.file("key.properties")
+   def keystoreProperties = new Properties()
+   keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
+
+   android {
+```
+
+1. Replace:
+```
+   buildTypes {
+       release {
+           // TODO: Add your own signing config for the release build.
+           // Signing with the debug keys for now, so `flutter run --release` works.
+           signingConfig signingConfigs.debug
+       }
+   }
+```
+   with:
+```
+   signingConfigs {
+       release {
+           keyAlias keystoreProperties['keyAlias']
+           keyPassword keystoreProperties['keyPassword']
+           storeFile file(keystoreProperties['storeFile'])
+           storePassword keystoreProperties['storePassword']
+       }
+   }
+   buildTypes {
+       release {
+           signingConfig signingConfigs.release
+       }
+   }
+```
+
+Release builds of your app will now automatically be signed.
+
+## Building a release APK
+
+This section describes how to build a release APK. If you completed the 
+signing steps in the previous section, the release APK will be signed.
+
+Using the command line:
+
+1. `cd <app dir>` (replace `<app dir>` with your application's directory).
+1. Run `flutter build apk`.
+
+The release APK for your app is created in `<app dir>/android/app/build/outputs/app.apk`.
```

Comment:
We might want to use `<app dir>/android/app/build/outputs/apk/app-release.apk` here. If they do a `flutter run` afterwards, then `app.apk` will be overwritten, but `app-release.apk` will not.